### PR TITLE
Clean cabal.project file

### PR DESCRIPTION
**Pull request by coot**
* consolidate `constraints`
* organise `package` options together

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -45,6 +37,21 @@ package ouroboros-network
...
+  type: git
+  location: https://github.com/input-output-hk/cardano-prelude
+  tag: 9352518fd9f957ab5e9611a2e27f495339c597d1
+  subdir: .
+
+source-repository-package
+  type: git
+  location: https://github.com/input-output-hk/cardano-prelude
+  tag: 9352518fd9f957ab5e9611a2e27f495339c597d1
+  subdir: .
```
Comment (mrBliss) [inline cabal.project]: Twice the same repo + subdir

Comment (coot) [reply cabal.project]: Good catch, I carried this from the current `cabal.project`.

Comment (coot) [reply cabal.project]: It's even repeated three times :D

**Pull request merged**